ui/app/helpers/utils/token-util.js

```diff
@@ -200,3 +201,30 @@ export function getFormattedTokenFiatAmount (
   })
   return `${formatCurrency(currentTokenInFiat, currentCurrency)} ${currentCurrency.toUpperCase()}`
 }
+
+/**
+ * Given a transaction receipt for a contract transaction that involves a token transfer to a given account,
+ * returns the amount of tokens transferred.
+ *
+ * @param {object} [txReceipt] - The transaction receipt object, as can be found in any standard MetaMask txMeta object
+ * @param {string} tokenAddres - The hex address of the token being transferred
+ * @param {string} receiverAddress - The address of the account being transferred to
+ * @param {number} [tokenDecimals] - The decimals of the token being transfered
+ * @returns {string|null} The amount of tokens transferred as a decimal number
+ */
+export function getTokensRecivedFromTxReceipt (txReceipt, tokenAddress, receiverAddress, tokenDecimals) {
+  const txReceiptLogs = txReceipt?.logs
+  if (txReceiptLogs && txReceipt?.status !== '0x0') {
+    const tokenTransferLog = txReceiptLogs.find((txReceiptLog) => {
+      const isTokenTransfer = txReceiptLog.topics && txReceiptLog.topics[0] === TOKEN_TRANSFER_LOG_TOPIC_HASH
+      const isTransferFromGivenToken = txReceiptLog.address === tokenAddress
+      const isTransferToReceiverAddress = txReceiptLog?.topics[2]?.match(receiverAddress.slice(2))
```

Comment:
Why use `.match` here, and slice off the prefix? :thinking:

Reply:
`match` isn't case-insensitive, so this depends upon the address being in the same case (perhaps it should be documented which case that is? I'm not sure what to expect here). So I don't know what utility using the `receiverAddress` as a regex has :thinking: 

This would be simpler:
```suggestion
      const isTransferToReceiverAddress = txReceiptLog?.topics[2] === receiverAddress.slice(2)
```

That is assuming that the topic does lack the `0x` prefix.

Reply:
Oh I see, the value is zero-padded. That makes sense.

Reply:
Perhaps this would be an improvement? A more explicit regex, and a comment:
```suggestion
      // the receiver address topic log is zero-padded
      const isTransferToReceiverAddress = txReceiptLog?.topics[2]?.match(new RegExp(`^0x0+${receiverAddress.slice(2)}$`)
```

Not blocking, just a suggestion.